### ISSUE TITLE
add logging utility features

### ISSUE THREAD

**Issue by quoeamaster** [quoeamaster/queutil#1]
**logging** is essential to all kinds of systems, a simple yet flexible logging architecture could solve things out in an efficient way.

Features involved:
* able to add logger(s) to the system; each logger would help to log down messages in a particular scenario (e.g. rolling-file logging, rdbms logging, console logging)
* provide lifecycle hooks / apis to add necessary resource claiming and resource release operations

**Comment by quoeamaster** [comment]
implementation => 

* ILogger interface = interface wrapping the functionalities of a logger
    * ***Write*** method to write out a message to a particular destionation
    * ***Name*** method returning the logger's name; use this name during the FlexLogger.WriteWithOptions() to indicate which logger implementation should be employed
    * ***Release*** method to release resources


* FlexLogger struct = the logger object for performing logging operations; 
    * ***AddLogger*** method is provided to add new ILogger interface implementation
    * ***Write*** method is provided for message logging (also acting as a **Writer** interface)
    * ***WriteWithOptions*** indicating which logger should be employed for the logging operation

```golang
    logger = queutil.NewFlexLogger()
    // added 2 loggers
    logger.AddLogger(queutil.NewRollingFileLogger(logFilepath, 1, 2, 2, true))
    logger.AddLogger(queutil.NewConsoleLogger())

    ...
    // basic write
    iWrote, err := logger.Write([]byte(message + "\n"))
    if err != nil {
        return err
    }
    fmt.Printf("success in logging, wrote '%v' bytes of data\n", iWrote)

    ... writeWithOptions
    optionMap := make(map[string]bool)
    optionMap["consoleLogger"] = true
    optionMap["rollingFileLogger"] = true
    _, err = loggerWithOptions.WriteWithOptions([]byte(message + "\n"), optionMap)
```

**Comment by quoeamaster** [comment]
feature released at **v0.1.1**
